### ISSUE TITLE
Tag allowed_values order difference results in repeated apply

### ISSUE THREAD

**Issue by chrisweis** [Snowflake-Labs/terraform-provider-snowflake#1806]
**Provider Version**

v0.64.0

**Terraform Version**

v1.4.6

**Describe the bug**

Somehow the order of the "allowed_values" property of a snowflake_tag resource is resulting in a repetitive modification upon "snowflake apply"

Screenshot for clarification:
![image](https://github.com/Snowflake-Labs/terraform-provider-snowflake/assets/13931731/f11fe9ff-27c3-4830-9805-256970f74ef7)

**Expected behavior**

I would expect that either:
1. It would understand that the allowed_values are identical even with a different order, or
2. It would succeed in modifying the order of values to resolve the needed modification.

My preference would be #2 (if it doesn't harm tag associations) and #1 if tag associations would be harmed.

**Code samples and commands**

Please add code examples and commands that were run to cause the problem.

My code:
```
resource "snowflake_tag" "controller" {
  name           = "controller"
  database       = snowflake_database.DEV_DW_DB.name
  schema         = snowflake_schema.DBT_STAGING.name
  allowed_values = ["terraform","dbt","manual"]
}
```
Snowflake's state:
```
resource "snowflake_tag" "controller" {
  name           = "controller"
  database       = snowflake_database.DEV_DW_DB.name
  schema         = snowflake_schema.DBT_STAGING.name
  allowed_values = ["dbt","manual","terraform"]
}
```

Oddly, when I do a `terraform destroy` and then `terraform apply` it still doesn't resolve this order difference, almost as those the order is being cached somewhere (possibly in Snowflake?)

**Comment by culpgrant** [comment]
This is also similar to #1498

**Comment by culpgrant** [comment]
@chrisweis I do have an unrelated question but are you tagging objects based on what system is creating them? I have been thinking about doing it because I think it would be easier to identify what owns an object.

**Comment by chrisweis** [comment]
Hi @culpgrant yes, exactly!  Ialso have other systems that control schema content (FiveTran, Segment, etc).  I was experimenting with tags for tracking where object control originates, but I'm starting to think that Snowflake ownership is probably enough (ex: TERRAFORM_ROLE is owner of the objects it controls while DBT_ROLE is owner of other objects it controls).  I'm working through the boundary between Terraform and dbt (especially for grants).

**Comment by culpgrant** [comment]
Yes interesting! I had the same idea and then decided that seeing the owner in Snowflake was enough. 
I think the tags are a good idea though.

**Comment by sfc-gh-jmichalak** [comment]
Hi @chrisweis @culpgrant   👋 

We've just released v0.99.0 ([release](https://github.com/Snowflake-Labs/terraform-provider-snowflake/releases/tag/v0.99.0), [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0980--v0990)) in which we reworked tags. This issue should be fixed. Please upgrade to this version and let us know if you have any issues.

**Comment by sfc-gh-jmichalak** [comment]
Closing due to inactivity.
